### PR TITLE
Api800/scope

### PR DESCRIPTION
**Pull request by supreethpy**
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: ov/scope.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	URI  utils.Nstring `json:"uri,omitempty"`  // "uri": "/rest/scopess/e2f0031b-52bd-4223-9ac1-d91cb519d548"
+}
+
+type ScopeList struct {
+	Total       int           `json:"total,omitempty"`       // "total": 1,
+	Count       int           `json:"count,omitempty"`       // "count": 1,
+	Start       int           `json:"start,omitempty"`       // "start": 0,
+	PrevPageURI utils.Nstring `json:"prevPageUri,omitempty"` // "prevPageUri": null,
+	NextPageURI utils.Nstring `json:"nextPageUri,omitempty"` // "nextPageUri": null,
+	URI         utils.Nstring `json:"uri,omitempty"`         // "uri": "/rest/server-profiles?filter=connectionTemplateUri%20matches%7769cae0-b680-435b-9b87-9b864c81657fsort=name:asc"
```
Comment (soodpr) [inline ov/scope.go]: commented uri should be scope specific

File: ov/scope.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	"github.com/HewlettPackard/oneview-golang/rest"
+	"github.com/HewlettPackard/oneview-golang/utils"
+	"github.com/docker/machine/libmachine/log"
+)
+
+type Scope struct {
+	Description utils.Nstring `json:"description,omitempty"` // "description": "Scope 1",
+	//Modified              string        `json:"modified,omitempty"`              // "modified": "20150831T154835.250Z",
+	Name string `json:"name,omitempty"` // "name": "Ethernet Network 1",
+	//State                 string        `json:"state,omitempty"`                 // "state": "Normal",
```
Comment (soodpr) [inline ov/scope.go]: remove comment out fields if they are not required

File: ov/scope.go (hunk 1 above)
Comment (supreethpy) [reply ov/scope.go]: Updated as per the comment.

File: ov/scope.go (hunk 2 above)
Comment (supreethpy) [reply ov/scope.go]: updated the structure members

File: ov/scope.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+	)
+	scps, err := c.GetScopes(fmt.Sprintf("name matches '%s'", name), "name:asc")
+	if scps.Total > 0 {
+		return scps.Members[0], err
+	} else {
+		return scp, err
+	}
+}
+
+func (c *OVClient) GetScopes(query string, sort string) (ScopeList, error) {
```
Comment (jyothis-gireesan) [inline ov/scope.go]: Please support all the query parameters.

File: examples/scope.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		err = ovc.UpdateScope(new_scp)
+
+		if err != nil {
+			fmt.Println("#.................... Scope Updation failed ...........#")
+			panic(err)
+		} else {
+			fmt.Println("#.................... Scope after Updating ...........#")
+		}
+	}
+	up_list, err := ovc.GetScopes("", sort)
```
Comment (soodpr) [inline examples/scope.go]: add count and start params as well as per document

File: ov/scope.go (hunk 3 above)
Comment (soodpr) [inline ov/scope.go]: query param name should be more specific here..

File: examples/scope.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+		os.Getenv("ONEVIEW_OV_DOMAIN"),
+		os.Getenv("ONEVIEW_OV_ENDPOINT"),
+		false,
+		800,
+		"*")
+
+	fmt.Println("#................... Scope by Name ...............#")
+	scp, scperr := ovc.GetScopeByName(scp_name)
+	if scperr != nil {
+		panic(scperr)
```
Comment (jyothis-gireesan) [inline examples/scope.go]: We can remove the panic, print the error and move ahead to the next line.

File: ov/scope.go [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+	Name          string           `json:"name,omitempty"`          // "name": "updated-SD3",
+	State         string           `json:"state,omitempty"`         // "state": "null",
+	Status        string           `json:"status,omitempty"`        // "status": "null",
+	Type          string           `json:"type,omitempty"`          // "type": "scopesV3",
+	URI           utils.Nstring    `json:"uri,omitempty"`           // "uri": "/rest/scopes/7f658031-c942-4336-be7a-67957cf20ba2"
+	ExtAttributes *ExtraAttributes `json:"extAttributes,omitempty"` //{}
+	ApplianceId   string           `json:"applianceId,omitempty"`   // "category": "scopes",
+	Category      string           `json:"category,omitempty"`      // "category": "scopes",
+	Created       string           `json:"created,omitempty"`       // "created": "2018-12-13T10:05:35.745Z",
+	ETAG          string           `json:"eTag,omitempty"`          // "eTag": "\"2018-12-13T10:24:25.267Z/2018-12-13T10:24:25.267Z\"",
```
Comment (jyothis-gireesan) [inline ov/scope.go]: Correct the example Etag

Comment (supreethpy) [reply ov/scope.go]: Updated

File: ov/scope.go (hunk 3 above)
Comment (supreethpy) [reply ov/scope.go]: added all the all the query parameters.

File: examples/scope.go (hunk 4 above)
Comment (supreethpy) [reply examples/scope.go]: Done

File: ov/scope.go (hunk 3 above)
Comment (supreethpy) [reply ov/scope.go]: in the API reference doc parameter by name query is there .

Comment (soodpr) [reply ov/scope.go]: you can ignore then

**Pull request merged**